- Disable xtest using xorg.conf.d (much better than previous method of altering
  the Xorg command line in the configuration files of display managers, which
  does doesn't even work anymore)
- Remove code for: 
  * other Unix-like operating systems
  * modifying: 
    - /etc/security/console.perms.d/50-default.perms - which has been removed
      since Fedora 10
    - /etc/logindevperm
    - /etc/X11/XF86Config
    - /etc/X11/gdm/gdm.conf
    - /etc/opt/gnome/gdm/gdm.conf
    - /etc/inittab - Fedora uses systemd.
  * removing /etc/modprobe.d/virtualgl (created by previous versions of 
    VirtualGL, which weren't even packaged for Fedora)
  * enabling XTest using gdm, kdm, or xdm
- Fix path: 
  * /bin/sh -> /bin/bash
  * /etc/modprobe.d/nvidia -> /etc/modprobe.d/nvidia.conf
  * /etc/opt/VirtualGL/ -> /etc/VirtualGL/
  * /etc/kde4/ -> /etc/kde/
- Do not add vglusers group. This is handled in the %post scriptlet
diff -Nru VirtualGL-2.3.orig/rr/vglserver_config VirtualGL-2.3/rr/vglserver_config
--- VirtualGL-2.3.orig/rr/vglserver_config	2012-02-25 00:12:07.914553456 -0500
+++ VirtualGL-2.3/rr/vglserver_config	2012-02-26 14:22:26.034754073 -0500
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # Copyright (C)2007 Sun Microsystems, Inc.
 # Copyright (C)2009-2011 D. R. Commander
@@ -179,53 +179,23 @@
 }
 
 disablextest() {
-	if [ -z "$1" ]; then
-		echo USAGE: disablextest {filename}
-		return
-	fi
-	backup $1
-	echo ... Disabling XTEST extension in $1 ...
+	backup /etc/X11/xorg.conf.d/00-disable-xtest.conf
+	echo ... Disabling XTEST extension ...
 	TMPFILE=`maketemp vglserver_config_7`
-	sed -e "/\/X11\/X\>/s/[ #\t]*-tst//g" -e "/\/bin\/X\>/s/[ #\t]*-tst//g" -e "/\/bin\/Xorg\>/s/[ #\t]*-tst//g" -e "/\/bin\/Xsun\>/s/[ #\t]*-tst//g" -e "/\/bin\/Xserver\>/s/[ #\t]*-tst//g" $1 >$TMPFILE && cp $TMPFILE $1 && (
-		sed -e "/\/X11\/X\>/s/$/\ -tst/g" -e "/\/bin\/X\>/s/$/\ -tst/g" -e "/\/bin\/Xorg\>/s/$/\ -tst/g" -e "/\/bin\/Xsun\>/s/$/\ -tst/g" -e "/\/bin\/Xserver\>/s/$/\ -tst/g" $1 >$TMPFILE && cp $TMPFILE $1
-	)
-	if [ -f $TMPFILE ]; then rm $TMPFILE; fi
-}
-
-disablextestkdm() {
-	if [ -z "$1" ]; then
-		echo USAGE: disablextestkdm {filename}
-		return
-	fi
-	backup $1
-	echo ... Disabling XTEST extension in $1 ...
-	TMPFILE=`maketemp vglserver_config_12`
-	sed -e "/ServerArgsLocal\>/s/[ #\t]*-tst//g" $1 >$TMPFILE && cp $TMPFILE $1 && (
-		sed -e "/ServerArgsLocal\>/s/$/\ -tst/g" $1 >$TMPFILE && cp $TMPFILE $1
-	)
-	if [ -f $TMPFILE ]; then rm $TMPFILE; fi
+	cat > ${TMPFILE} << EOF
+Section "Extensions"
+	Option "XTEST" "Disable"
+EndSection
+EOF
+	cp ${TMPFILE} /etc/X11/xorg.conf.d/00-disable-xtest.conf
+	if [ -f ${TMPFILE} ]; then rm ${TMPFILE}; fi
 }
 
 enablextest() {
-	if [ -z "$1" ]; then
-		echo USAGE: enablextest {filename}
-		return
+	echo ... Enabling XTEST extension ...
+	if [ -f /etc/X11/xorg.conf.d/00-disable-xtest.conf ]; then
+		rm /etc/X11/xorg.conf.d/00-disable-xtest.conf
 	fi
-	echo ... Enabling XTEST extension in $1 ...
-	TMPFILE=`maketemp vglserver_config_8`
-	sed -e "/\/X11\/X\>/s/[ #\t]*-tst//g" -e "/\/bin\/X\>/s/[ #\t]*-tst//g" -e "/\/bin\/Xorg\>/s/[ #\t]*-tst//g" -e "/\/bin\/Xsun\>/s/[ #\t]*-tst//g" -e "/\/bin\/Xserver\>/s/[ #\t]*-tst//g" $1 >$TMPFILE && cp $TMPFILE $1
-	if [ -f $TMPFILE ]; then rm $TMPFILE; fi
-}
-
-enablextestkdm() {
-	if [ -z "$1" ]; then
-		echo USAGE: enablextestkdm {filename}
-		return
-	fi
-	echo ... Enabling XTEST extension in $1 ...
-	TMPFILE=`maketemp vglserver_config_11`
-	sed -e "/ServerArgsLocal\>/s/[ #\t]*-tst//g" $1 >$TMPFILE && cp $TMPFILE $1
-	if [ -f $TMPFILE ]; then rm $TMPFILE; fi
 }
 
 disallowgdmtcp() {
@@ -261,118 +231,73 @@
 }
 
 unconfigdev() {
-	if [ "$UNAME_S" = "SunOS" ]; then
-		echo ... Modifying /etc/logindevperm to enable automatic permissions for
-		echo "    /dev/fbs/* ..."
-		uncommentline /etc/logindevperm "\/dev\/console.*\/dev\/fbs"
-	elif [ "$UNAME_S" = "Linux" ]; then
-		if [ -f /etc/security/console.perms ]; then
-			echo ... Modifying /etc/security/console.perms to enable automatic permissions
-			echo "    for DRI devices ..."
-			uncommentline /etc/security/console.perms "\<dri\>"
-		fi
-		if [ -f /etc/security/console.perms.d/50-default.perms ]; then
-			echo ... Modifying /etc/security/console.perms.d/50-default.perms to enable automatic permissions
-			echo "    for DRI devices ..."
-			uncommentline /etc/security/console.perms.d/50-default.perms "\<dri\>"
-		fi
-		if [ -f /etc/logindevperm ]; then
-			echo ... Modifying /etc/logindevperm to enable automatic permissions for
-			echo "    /dev/nvidia* ..."
-			uncommentline /etc/logindevperm "\/dev\/nvidia"
-		fi
-		if [ -f /etc/modprobe.d/virtualgl.conf -o -f /etc/modprobe.d/virtualgl ]; then
-			echo ... Removing /etc/modprobe.d/virtualgl.conf to restore default permissions for
-			echo "    /dev/nvidia* ..."
-			rm -f /etc/modprobe.d/virtualgl
-			rm -f /etc/modprobe.d/virtualgl.conf
-			$LSMOD | grep -q nvidia >/dev/null 2>/dev/null
-			if [ $? = 0 ]; then
-				echo ... Attempting to remove nvidia module from memory so device permissions
-				echo "    will be reloaded ..."
-				$RMMOD nvidia || RMMODNEEDED=1
-			fi
-
+	if [ -f /etc/security/console.perms ]; then
+		echo ... Modifying /etc/security/console.perms to enable automatic permissions
+		echo "    for DRI devices ..."
+		uncommentline /etc/security/console.perms "\<dri\>"
+	fi
+	if [ -f /etc/modprobe.d/virtualgl.conf ]; then
+		echo ... Removing /etc/modprobe.d/virtualgl.conf to restore default permissions for
+		echo "    /dev/nvidia* ..."
+		rm -f /etc/modprobe.d/virtualgl.conf
+		$LSMOD | grep -q nvidia >/dev/null 2>/dev/null
+		if [ $? = 0 ]; then
+			echo ... Attempting to remove nvidia module from memory so device permissions
+			echo "    will be reloaded ..."
+			$RMMOD nvidia || RMMODNEEDED=1
 		fi
 	fi
 }
 
 configdev() {
-	if [ "$UNAME_S" = "SunOS" ]; then
-		echo ... Modifying /etc/logindevperm to disable automatic permissions for
-		echo "    /dev/fbs/* ..."
-		commentline /etc/logindevperm "\/dev\/console.*\/dev\/fbs"
-		chmod 755 /dev/fbs
-		if [ "$FBDEVVGLUSERSONLY" = "1" ]; then
-			echo ... Granting write permission to /dev/fbs/\* for vglusers group ...
-			chmod 660 /dev/fbs/*
-			chown root:vglusers /dev/fbs/*
-		else
-			echo ... Granting write permission to /dev/fbs/\* for all users ...
-			chmod 666 /dev/fbs/*
-			chown root:root /dev/fbs/*
-		fi
-	elif [ "$UNAME_S" = "Linux" ]; then
-		if [ -f /etc/security/console.perms ]; then
-			echo ... Modifying /etc/security/console.perms to disable automatic permissions
-			echo "    for DRI devices ..."
-			commentline /etc/security/console.perms "\<dri\>"
-		fi
-		if [ -f /etc/security/console.perms.d/50-default.perms ]; then
-			echo ... Modifying /etc/security/console.perms.d/50-default.perms to disable automatic permissions
-			echo "    for DRI devices ..."
-			commentline /etc/security/console.perms.d/50-default.perms "\<dri\>"
-		fi
-		if [ -f /etc/logindevperm ]; then
-			echo ... Modifying /etc/logindevperm to disable automatic permissions for
-			echo "    /dev/nvidia* ..."
-			commentline /etc/logindevperm "\/dev\/nvidia"
-		fi
-		if [ -d /etc/modprobe.d ]; then
-			if [ -f /etc/modprobe.d/nvidia ]; then
-				echo ... Modifying /etc/modprobe.d/nvidia to set requested permissions for
+	if [ -f /etc/security/console.perms ]; then
+		echo ... Modifying /etc/security/console.perms to disable automatic permissions
+		echo "    for DRI devices ..."
+		commentline /etc/security/console.perms "\<dri\>"
+	fi
+	if [ -d /etc/modprobe.d ]; then
+		if [ -f /etc/modprobe.d/nvidia.conf ]; then
+			echo ... Modifying /etc/modprobe.d/nvidia.conf to set requested permissions for
+		else
+			echo ... Creating /etc/modprobe.d/virtualgl.conf to set requested permissions for
+		fi
+		echo "    /dev/nvidia* ..."
+		VGLUSERSGID=`grep vglusers /etc/group | cut -f3 -d:`
+		if [ -f /etc/modprobe.d/nvidia.conf ]; then
+			if [ "$FBDEVVGLUSERSONLY" = "1" -a ! "$VGLUSERSGID" = "" ]; then
+				replaceline /etc/modprobe.d/nvidia.conf "options nvidia" "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=$VGLUSERSGID NVreg_DeviceFileMode=0660"
 			else
-				echo ... Creating /etc/modprobe.d/virtualgl.conf to set requested permissions for
+				replaceline /etc/modprobe.d/nvidia.conf "options nvidia" "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=0 NVreg_DeviceFileMode=0666"
 			fi
-			echo "    /dev/nvidia* ..."
-			VGLUSERSGID=`grep vglusers /etc/group | cut -f3 -d:`
-			if [ -f /etc/modprobe.d/nvidia ]; then
-				if [ "$FBDEVVGLUSERSONLY" = "1" -a ! "$VGLUSERSGID" = "" ]; then
-					replaceline /etc/modprobe.d/nvidia "options nvidia" "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=$VGLUSERSGID NVreg_DeviceFileMode=0660"
-				else
-					replaceline /etc/modprobe.d/nvidia "options nvidia" "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=0 NVreg_DeviceFileMode=0666"
-				fi
+		else
+			if [ "$FBDEVVGLUSERSONLY" = "1" -a ! "$VGLUSERSGID" = "" ]; then
+				echo "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=$VGLUSERSGID NVreg_DeviceFileMode=0660" >/etc/modprobe.d/virtualgl.conf
 			else
-				if [ "$FBDEVVGLUSERSONLY" = "1" -a ! "$VGLUSERSGID" = "" ]; then
-					echo "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=$VGLUSERSGID NVreg_DeviceFileMode=0660" >/etc/modprobe.d/virtualgl.conf
-				else
-					echo "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=0 NVreg_DeviceFileMode=0666" >/etc/modprobe.d/virtualgl.conf
-				fi
-				chmod 644 /etc/modprobe.d/virtualgl.conf
-			fi
-			$LSMOD | grep -q nvidia >/dev/null 2>/dev/null
-			if [ $? = 0 ]; then
-				echo ... Attempting to remove nvidia module from memory so device permissions
-				echo "    will be reloaded ..."
-				$RMMOD nvidia || RMMODNEEDED=1
+				echo "options nvidia NVreg_DeviceFileUID=0 NVreg_DeviceFileGID=0 NVreg_DeviceFileMode=0666" >/etc/modprobe.d/virtualgl.conf
 			fi
+			chmod 644 /etc/modprobe.d/virtualgl.conf
 		fi
-		if [ "$FBDEVVGLUSERSONLY" = "1" ]; then
-			if [ -e /dev/nvidia0 -o -e /dev/nvidiactl ]; then
-				echo ... Granting write permission to /dev/nvidia* for vglusers group ...
-				chmod 660 /dev/nvidia*
-				chown root:vglusers /dev/nvidia*
-			fi
-		else
-			if [ -e /dev/nvidia0 -o -e /dev/nvidiactl ]; then
-				echo ... Granting write permission to /dev/nvidia* for all users ...
-				chmod 666 /dev/nvidia*
-				chown root:root /dev/nvidia*
-			fi
+		$LSMOD | grep -q nvidia >/dev/null 2>/dev/null
+		if [ $? = 0 ]; then
+			echo ... Attempting to remove nvidia module from memory so device permissions
+			echo "    will be reloaded ..."
+			$RMMOD nvidia || RMMODNEEDED=1
+		fi
+	fi
+	if [ "$FBDEVVGLUSERSONLY" = "1" ]; then
+		if [ -e /dev/nvidia0 -o -e /dev/nvidiactl ]; then
+			echo ... Granting write permission to /dev/nvidia* for vglusers group ...
+			chmod 660 /dev/nvidia*
+			chown root:vglusers /dev/nvidia*
+		fi
+	else
+		if [ -e /dev/nvidia0 -o -e /dev/nvidiactl ]; then
+			echo ... Granting write permission to /dev/nvidia* for all users ...
+			chmod 666 /dev/nvidia*
+			chown root:root /dev/nvidia*
 		fi
-		setdripermissions /etc/X11/xorg.conf
-		setdripermissions /etc/X11/XF86Config
 	fi
+	setdripermissions /etc/X11/xorg.conf
 }
 
 ynprompt() {
@@ -395,9 +320,9 @@
 
 unconfigglx() {
 	echo ... Removing /etc/opt/VirtualGL directory ...
-	if [ -d /etc/opt/VirtualGL ]; then
-		rm /etc/opt/VirtualGL/* 2>/dev/null
-		rmdir /etc/opt/VirtualGL
+	if [ -d /etc/VirtualGL ]; then
+		rm /etc/VirtualGL/* 2>/dev/null
+		rmdir /etc/VirtualGL
 	fi
 
 	if [ -f /etc/X11/xdm/Xsetup_0 -a ! -h /etc/X11/xdm/Xsetup_0 ]; then
@@ -407,86 +332,15 @@
 			delvglgenkey /etc/X11/xdm/Xsetup
 		fi
 	fi
-	if [ -f /etc/X11/gdm/Init/:0 ]; then
-		if [ ! -h /etc/X11/gdm/Init/:0 ]; then
-			delvglgenkey /etc/X11/gdm/Init/:0
-		fi
-	else
-		if [ -f /etc/X11/gdm/Init/Default -a ! -h /etc/X11/gdm/Init/Default ]; then
-			delvglgenkey /etc/X11/gdm/Init/Default
-		fi
-	fi
-	if [ -f /etc/X11/gdm/SunRayInit/Default ]; then
-		delvglgenkey /etc/X11/gdm/SunRayInit/Default
-	fi
 	if [ -f /etc/gdm/Init/Default -a ! -h /etc/gdm/Init/Default ]; then
 		delvglgenkey /etc/gdm/Init/Default
 	fi
-	if [ -f /etc/opt/gnome/gdm/Init/Default -a ! -h /etc/opt/gnome/gdm/Init/Default ]; then
-		delvglgenkey /etc/opt/gnome/gdm/Init/Default
-	fi
-	if [ -f /etc/kde4/kdm/Xsetup -a ! -h /etc/kde4/kdm/Xsetup ]; then
-		delvglgenkey /etc/kde4/kdm/Xsetup
+	if [ -f /etc/kde/kdm/Xsetup -a ! -h /etc/kde/kdm/Xsetup ]; then
+		delvglgenkey /etc/kde/kdm/Xsetup
 	fi
 
-	if [ "$UNAME_S" = "SunOS" ]; then
-		if [ -f /etc/dt/config/Xsetup ]; then
-			delvglgenkey /etc/dt/config/Xsetup
-		fi
-		if [ -f /etc/dt/config/Xconfig ]; then
-			echo ... Restoring default value of Dtlogin\*grabServer in ...
-			echo "...     /etc/dt/config/Xconfig ..."
-			commentline /etc/dt/config/Xconfig Dtlogin\\*grabServer
-		fi
-		if [ -f /etc/dt/config/Xconfig.SUNWut.prototype ]; then
-			echo ... Restoring default value of Dtlogin\*grabServer in ...
-			echo "...     /etc/dt/config/Xconfig.SUNWut.prototype ..."
-			commentline /etc/dt/config/Xconfig.SUNWut.prototype Dtlogin\\*grabServer
-		fi
-	fi
-
-	if [ -f /etc/X11/xdm/Xservers -a ! -h /etc/X11/xdm/Xservers ]; then
-		enablextest /etc/X11/xdm/Xservers
-	fi
-	if [ -f /etc/X11/gdm/gdm.conf -a ! -h /etc/X11/gdm/gdm.conf ]; then
-		enablextest /etc/X11/gdm/gdm.conf
-	fi
-	if [ -f /etc/gdm/custom.conf -a ! -h /etc/gdm/custom.conf ]; then
-		enablextest /etc/gdm/custom.conf
-	fi
-	if [ -f /etc/gdm/gdm.conf -a ! -h /etc/gdm/gdm.conf ]; then
-		enablextest /etc/gdm/gdm.conf
-	fi
-	if [ -f /etc/gdm/gdm-cdd.conf ]; then
-		enablextest /etc/gdm/gdm-cdd.conf
-	fi
-	if [ -f /etc/X11/gdm/custom.conf -a ! -h /etc/X11/gdm/custom.conf ]; then
-		enablextest /etc/X11/gdm/custom.conf
-	fi
-	if [ -f /etc/opt/gnome/gdm/gdm.conf -a ! -h /etc/opt/gnome/gdm/gdm.conf ]; then
-		enablextest /etc/opt/gnome/gdm/gdm.conf
-	fi
-	if [ -f /etc/opt/kde3/share/config/kdm/Xservers -a ! -h /etc/opt/kde3/share/config/kdm/Xservers ]; then
-		enablextest /etc/opt/kde3/share/config/kdm/Xservers
-	fi
-	if [ -f /etc/kde4/kdm/kdmrc -a ! -h /etc/kde4/kdm/kdmrc ]; then
-		enablextestkdm /etc/kde4/kdm/kdmrc
-	fi
-	if [ -f /etc/kde/kdm/kdmrc -a ! -h /etc/kde/kdm/kdmrc ]; then
-		enablextestkdm /etc/kde/kdm/kdmrc
-	fi
-	if [ "$UNAME_S" = "SunOS" ]; then
-		if [ -f /etc/dt/config/Xservers ]; then
-			enablextest /etc/dt/config/Xservers
-		fi
-		if [ -x /usr/sbin/svccfg ]; then
-			/usr/sbin/svccfg -s application/x11/x11-server delpropvalue options/server_args "*-tst*" 2>&1 >/dev/null
-		fi
-	fi
+	enablextest
 
-	if [ -f /etc/X11/gdm/gdm.conf -a ! -h /etc/X11/gdm/gdm.conf ]; then
-		disallowgdmtcp /etc/X11/gdm/gdm.conf
-	fi
 	if [ -f /etc/gdm/custom.conf -a ! -h /etc/gdm/custom.conf ]; then
 		disallowgdmtcp /etc/gdm/custom.conf
 	fi
@@ -496,12 +350,6 @@
 	if [ -f /etc/gdm/gdm-cdd.conf ]; then
 		disallowgdmtcp /etc/gdm/gdm-cdd.conf
 	fi
-	if [ -f /etc/X11/gdm/custom.conf -a ! -h /etc/X11/gdm/custom.conf ]; then
-		disallowgdmtcp /etc/X11/gdm/custom.conf
-	fi
-	if [ -f /etc/opt/gnome/gdm/gdm.conf -a ! -h /etc/opt/gnome/gdm/gdm.conf ]; then
-		disallowgdmtcp /etc/opt/gnome/gdm/gdm.conf
-	fi
 
 	unconfigdev
 
@@ -529,20 +377,13 @@
 	fi
 
 	if [ "$VGLUSERSONLY" = "1" ]; then
-		echo ... Creating vglusers group ...
-		if [ ! "$GID" = "" ]; then
-			$GROUPADD -g $GID vglusers || echo "Could not add vglusers group (probably because it already exists.)"
-		else
-			$GROUPADD vglusers || echo "Could not add vglusers group (probably because it already exists.)"
-		fi
-
-		if [ ! -d /etc/opt/VirtualGL ]; then
-			echo ... Creating /etc/opt/VirtualGL/ ...
-			mkdir -p /etc/opt/VirtualGL
-		fi
-		echo ... Granting read permission to /etc/opt/VirtualGL/ for vglusers group ...
-		chown root:vglusers /etc/opt/VirtualGL
-		chmod 750 /etc/opt/VirtualGL
+		if [ ! -d /etc/VirtualGL ]; then
+			echo ... Creating /etc/VirtualGL/ ...
+			mkdir -p /etc/VirtualGL
+		fi
+		echo ... Granting read permission to /etc/VirtualGL/ for vglusers group ...
+		chown root:vglusers /etc/VirtualGL
+		chmod 750 /etc/VirtualGL
 	fi
 
 	configdev
@@ -554,172 +395,19 @@
 			addvglgenkey /etc/X11/xdm/Xsetup
 		fi
 	fi
-	if [ -f /etc/X11/gdm/Init/:0 ]; then
-		if [ ! -h /etc/X11/gdm/Init/:0 ]; then
-			addvglgenkey /etc/X11/gdm/Init/:0
-		fi
-	else
-		if [ -f /etc/X11/gdm/Init/Default -a ! -h /etc/X11/gdm/Init/Default ]; then
-			addvglgenkey /etc/X11/gdm/Init/Default
-		fi
-	fi
-	if [ -f /etc/X11/gdm/SunRayInit/Default ]; then
-		addvglgenkey /etc/X11/gdm/SunRayInit/Default
-	fi
 	if [ -f /etc/gdm/Init/Default -a ! -h /etc/gdm/Init/Default ]; then
 		addvglgenkey /etc/gdm/Init/Default
 	fi
-	if [ -f /etc/opt/gnome/gdm/Init/Default -a ! -h /etc/opt/gnome/gdm/Init/Default ]; then
-		addvglgenkey /etc/opt/gnome/gdm/Init/Default
-	fi
-	if [ -f /etc/kde4/kdm/Xsetup -a ! -h /etc/kde4/kdm/Xsetup ]; then
-		addvglgenkey /etc/kde4/kdm/Xsetup
+	if [ -f /etc/kde/kdm/Xsetup -a ! -h /etc/kde/kdm/Xsetup ]; then
+		addvglgenkey /etc/kde/kdm/Xsetup
 	fi
 		
-	if [ "$UNAME_S" = "SunOS" ]; then
-		if [ ! -d /etc/dt/config ]; then
-			echo ... Creating /etc/dt/config/ directory ...
-			mkdir -p /etc/dt/config
-			chmod 755 /etc/dt/config
-		fi
-
-		if [ ! -f /etc/dt/config/Xsetup ]; then
-			if [ -f /usr/dt/config/Xsetup ]; then
-				echo ... Copying /usr/dt/config/Xsetup to /etc/dt/config/Xsetup ...
-				cp /usr/dt/config/Xsetup /etc/dt/config/Xsetup
-			fi
-		else
-			backup /etc/dt/config/Xsetup
-		fi
-		if [ -f /etc/dt/config/Xsetup ]; then
-			addvglgenkey /etc/dt/config/Xsetup bottom
-		fi
-
-		if [ ! -f /etc/dt/config/Xconfig ]; then
-			if [ -f /usr/dt/config/Xconfig ]; then
-				echo ... Copying /usr/dt/config/Xconfig to /etc/dt/config/Xconfig ...
-				cp /usr/dt/config/Xconfig /etc/dt/config/Xconfig
-			fi
-		else
-			backup /etc/dt/config/Xconfig
-		fi
-		if [ -f /etc/dt/config/Xconfig ]; then
-			echo ... Setting Dtlogin\*grabServer to False in /etc/dt/config/Xconfig ...
-			replaceline /etc/dt/config/Xconfig Dtlogin\\*grabServer "Dtlogin\*grabServer: False"
-		fi
-		if [ -f /etc/dt/config/Xconfig.SUNWut.prototype ]; then
-			backup /etc/dt/config/Xconfig.SUNWut.prototype
-			echo ... Setting Dtlogin\*grabServer to False in /etc/dt/config/Xconfig.SUNWut.prototype ...
-			replaceline /etc/dt/config/Xconfig.SUNWut.prototype Dtlogin\\*grabServer "Dtlogin\*grabServer: False"
-		fi
-
-	fi
-
-	if [ -f /etc/X11/xdm/Xservers -a ! -h /etc/X11/xdm/Xservers ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/X11/xdm/Xservers
-		else
-			enablextest /etc/X11/xdm/Xservers
-		fi
-	fi
-	if [ -f /etc/X11/gdm/gdm.conf -a ! -h /etc/X11/gdm/gdm.conf ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/X11/gdm/gdm.conf
-		else
-			enablextest /etc/X11/gdm/gdm.conf
-		fi
-	fi
-	if [ -f /etc/gdm/custom.conf -a ! -h /etc/gdm/custom.conf ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/gdm/custom.conf
-		else
-			enablextest /etc/gdm/custom.conf
-		fi
-	fi
-	if [ -f /etc/gdm/gdm.conf -a ! -h /etc/gdm/gdm.conf ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/gdm/gdm.conf
-		else
-			enablextest /etc/gdm/gdm.conf
-		fi
-	fi
-	if [ -f /etc/gdm/gdm-cdd.conf ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/gdm/gdm-cdd.conf
-		else
-			enablextest /etc/gdm/gdm-cdd.conf
-		fi
-	fi
-	if [ -f /etc/X11/gdm/custom.conf -a ! -h /etc/X11/gdm/custom.conf ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/X11/gdm/custom.conf
-		else
-			enablextest /etc/X11/gdm/custom.conf
-		fi
-	fi
-	if [ -f /etc/opt/gnome/gdm/gdm.conf -a ! -h /etc/opt/gnome/gdm/gdm.conf ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/opt/gnome/gdm/gdm.conf
-		else
-			enablextest /etc/opt/gnome/gdm/gdm.conf
-		fi
-	fi
-	if [ -f /etc/opt/kde3/share/config/kdm/Xservers -a ! -h /etc/opt/kde3/share/config/kdm/Xservers ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextest /etc/opt/kde3/share/config/kdm/Xservers
-		else
-			enablextest /etc/opt/kde3/share/config/kdm/Xservers
-		fi
-	fi
-	if [ -f /etc/kde4/kdm/kdmrc -a ! -h /etc/kde4/kdm/kdmrc ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextestkdm /etc/kde4/kdm/kdmrc
-		else
-			enablextestkdm /etc/kde4/kdm/kdmrc
-		fi
-	fi
-	if [ -f /etc/kde/kdm/kdmrc -a ! -h /etc/kde/kdm/kdmrc ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			disablextestkdm /etc/kde/kdm/kdmrc
-		else
-			enablextestkdm /etc/kde/kdm/kdmrc
-		fi
-	fi
-	if [ "$UNAME_S" = "SunOS" ]; then
-		if [ "$DISABLEXTEST" = "1" ]; then
-			if [ ! -f /etc/dt/config/Xservers ]; then
-				if [ -f /usr/dt/config/Xservers ]; then
-		 			echo ... Copying /usr/dt/config/Xservers to /etc/dt/config/Xservers ...
-					cp /usr/dt/config/Xservers /etc/dt/config/Xservers
-				fi
-			fi
-			if [ -f /etc/dt/config/Xservers ]; then
-				disablextest /etc/dt/config/Xservers
-			fi
-			if [ -x /usr/sbin/svccfg ]; then
-				/usr/sbin/svccfg -s application/x11/x11-server delpropvalue options/server_args "*-tst*" 2>&1 >/dev/null
-				/usr/sbin/svccfg -s application/x11/x11-server addpropvalue options/server_args -tst 2>&1 >/dev/null
-			fi
-		else
-			if [ -f /etc/dt/config/Xservers ]; then
-				enablextest /etc/dt/config/Xservers
-			fi
-			if [ -x /usr/sbin/svccfg ]; then
-				/usr/sbin/svccfg -s application/x11/x11-server delpropvalue options/server_args "*-tst*" 2>&1 >/dev/null
-			fi
-		fi
-	fi
-
-	if [ "$UNAME_S" = "Linux" ]; then
-		if [ -f /etc/inittab ]; then
-			echo ... Setting default run level to 5 \(enabling graphical login prompt\) ...
-			replaceline /etc/inittab "id:[0-9]:initdefault:" "id:5:initdefault:"
-		fi
+	if [ "$DISABLEXTEST" = "1" ]; then
+		disablextest
+	else
+		enablextest
 	fi
 
-	if [ -f /etc/X11/gdm/gdm.conf -a ! -h /etc/X11/gdm/gdm.conf ]; then
-		disallowgdmtcp /etc/X11/gdm/gdm.conf
-	fi
 	if [ -f /etc/gdm/custom.conf -a ! -h /etc/gdm/custom.conf ]; then
 		disallowgdmtcp /etc/gdm/custom.conf
 	fi
@@ -729,12 +417,6 @@
 	if [ -f /etc/gdm/gdm-cdd.conf ]; then
 		disallowgdmtcp /etc/gdm/gdm-cdd.conf
 	fi
-	if [ -f /etc/X11/gdm/custom.conf -a ! -h /etc/X11/gdm/custom.conf ]; then
-		disallowgdmtcp /etc/X11/gdm/custom.conf
-	fi
-	if [ -f /etc/opt/gnome/gdm/gdm.conf -a ! -h /etc/opt/gnome/gdm/gdm.conf ]; then
-		disallowgdmtcp /etc/opt/gnome/gdm/gdm.conf
-	fi
 
 	echo
 	echo Done.  You must restart the display manager for the changes to take effect.
